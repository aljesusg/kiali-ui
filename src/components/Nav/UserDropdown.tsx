import * as React from 'react';
import { Dropdown, Icon, MenuItem } from 'patternfly-react';
import PropTypes from 'prop-types';
import store from '../../store/ConfigStore';
import * as UserAction from '../../actions/UserAction';

type UserProps = {
  dispatch?: any;
  handleAuthenticate: PropTypes.func;
};

type UserState = {
  username: string;
};

class UserDropdown extends React.Component<UserProps, UserState> {
  static contextTypes = {
    router: PropTypes.object
  };

  constructor(props: UserProps) {
    super(props);
    const actualState = store.getState() || {};
    let username = '';
    if (actualState['authentication']['user'] !== null) {
      username = actualState['authentication']['user']['username'];
    }

    // handle initial path from the browser
    this.state = {
      username: username
    };
  }

  userLogout = () => {
    store.dispatch(UserAction.logout());
    this.props.handleAuthenticate(false);
  };
  render() {
    return (
      <>
        <Dropdown componentClass="li" id="user">
          <Dropdown.Toggle useAnchor={true} className="nav-item-iconic">
            <Icon type="pf" name="user" /> {this.state.username}
          </Dropdown.Toggle>
          <Dropdown.Menu>
            <MenuItem id="usermenu_logout" onClick={() => this.userLogout()}>
              <Icon type="pf" name="key" /> Logout
            </MenuItem>
          </Dropdown.Menu>
        </Dropdown>
      </>
    );
  }
}

export default UserDropdown;
