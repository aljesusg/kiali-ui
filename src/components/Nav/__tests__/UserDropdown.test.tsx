import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { Dropdown, Icon } from 'patternfly-react';
import UserDropdown from '../UserDropdown';

describe('userDropdown test', () => {
  it('should render UserDropdown correctly', () => {
    sessionStorage.setItem('user', 'jdoe');
    const wrapper = shallow(<UserDropdown />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('should include the username with Icon ', () => {
    const username = 'jdoe';
    sessionStorage.setItem('user', username);
    const wrapper = mount(<UserDropdown />);
    expect(
      wrapper.find(Dropdown.Toggle).contains(username) &&
        wrapper.find(Dropdown.Toggle).contains(<Icon type="pf" name="user" />)
    ).toBeTruthy();
  });

  it('should include a logout option and remove session storage', () => {
    const username = 'jdoe';
    sessionStorage.setItem('user', username);
    const wrapper = mount(<UserDropdown />);
    expect(wrapper.find('#usermenu_logout').exists()).toBeTruthy();
    expect(wrapper.find('#usermenu_logout').contains(<Icon type="pf" name="key" />)).toBeTruthy();
    const elt = wrapper
      .find('#usermenu_logout')
      .find('SafeAnchor')
      .first();
    elt.simulate('click');
    expect(sessionStorage.getItem('user')).toBeNull();
  });
});
