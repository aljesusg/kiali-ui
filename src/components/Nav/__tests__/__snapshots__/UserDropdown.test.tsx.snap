// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userDropdown test should render UserDropdown correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserDropdown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Uncontrolled(Dropdown)
        componentClass="li"
        id="user"
      >
        <DropdownToggle
          bsClass="dropdown-toggle"
          bsRole="toggle"
          className="nav-item-iconic"
          open={false}
          useAnchor={true}
        >
          <Icon
            name="user"
            type="pf"
          />
           
          jdoe
        </DropdownToggle>
        <DropdownMenu
          bsClass="dropdown-menu"
          bsRole="menu"
          pullRight={false}
        >
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            id="usermenu_logout"
            onClick={[Function]}
          >
            <Icon
              name="key"
              type="pf"
            />
             Logout
          </MenuItem>
        </DropdownMenu>
      </Uncontrolled(Dropdown)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <DropdownToggle
            bsClass="dropdown-toggle"
            bsRole="toggle"
            className="nav-item-iconic"
            open={false}
            useAnchor={true}
          >
            <Icon
              name="user"
              type="pf"
            />
             
            jdoe
          </DropdownToggle>,
          <DropdownMenu
            bsClass="dropdown-menu"
            bsRole="menu"
            pullRight={false}
          >
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              header={false}
              id="usermenu_logout"
              onClick={[Function]}
            >
              <Icon
                name="key"
                type="pf"
              />
               Logout
            </MenuItem>
          </DropdownMenu>,
        ],
        "componentClass": "li",
        "id": "user",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "dropdown-toggle",
            "bsRole": "toggle",
            "children": Array [
              <Icon
                name="user"
                type="pf"
              />,
              " ",
              "jdoe",
            ],
            "className": "nav-item-iconic",
            "open": false,
            "useAnchor": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "name": "user",
                "type": "pf",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " ",
            "jdoe",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "dropdown-menu",
            "bsRole": "menu",
            "children": <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              header={false}
              id="usermenu_logout"
              onClick={[Function]}
            >
              <Icon
                name="key"
                type="pf"
              />
               Logout
            </MenuItem>,
            "pullRight": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "dropdown",
              "children": Array [
                <Icon
                  name="key"
                  type="pf"
                />,
                " Logout",
              ],
              "disabled": false,
              "divider": false,
              "header": false,
              "id": "usermenu_logout",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "name": "key",
                  "type": "pf",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              " Logout",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Uncontrolled(Dropdown)
          componentClass="li"
          id="user"
        >
          <DropdownToggle
            bsClass="dropdown-toggle"
            bsRole="toggle"
            className="nav-item-iconic"
            open={false}
            useAnchor={true}
          >
            <Icon
              name="user"
              type="pf"
            />
             
            jdoe
          </DropdownToggle>
          <DropdownMenu
            bsClass="dropdown-menu"
            bsRole="menu"
            pullRight={false}
          >
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              header={false}
              id="usermenu_logout"
              onClick={[Function]}
            >
              <Icon
                name="key"
                type="pf"
              />
               Logout
            </MenuItem>
          </DropdownMenu>
        </Uncontrolled(Dropdown)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <DropdownToggle
              bsClass="dropdown-toggle"
              bsRole="toggle"
              className="nav-item-iconic"
              open={false}
              useAnchor={true}
            >
              <Icon
                name="user"
                type="pf"
              />
               
              jdoe
            </DropdownToggle>,
            <DropdownMenu
              bsClass="dropdown-menu"
              bsRole="menu"
              pullRight={false}
            >
              <MenuItem
                bsClass="dropdown"
                disabled={false}
                divider={false}
                header={false}
                id="usermenu_logout"
                onClick={[Function]}
              >
                <Icon
                  name="key"
                  type="pf"
                />
                 Logout
              </MenuItem>
            </DropdownMenu>,
          ],
          "componentClass": "li",
          "id": "user",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "dropdown-toggle",
              "bsRole": "toggle",
              "children": Array [
                <Icon
                  name="user"
                  type="pf"
                />,
                " ",
                "jdoe",
              ],
              "className": "nav-item-iconic",
              "open": false,
              "useAnchor": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "name": "user",
                  "type": "pf",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              " ",
              "jdoe",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "dropdown-menu",
              "bsRole": "menu",
              "children": <MenuItem
                bsClass="dropdown"
                disabled={false}
                divider={false}
                header={false}
                id="usermenu_logout"
                onClick={[Function]}
              >
                <Icon
                  name="key"
                  type="pf"
                />
                 Logout
              </MenuItem>,
              "pullRight": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "dropdown",
                "children": Array [
                  <Icon
                    name="key"
                    type="pf"
                  />,
                  " Logout",
                ],
                "disabled": false,
                "divider": false,
                "header": false,
                "id": "usermenu_logout",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "name": "key",
                    "type": "pf",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                " Logout",
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
